presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -85,4 +99,53 @@ public static long getFileSize(Page statisticsPage, int position)
         }
         return Optional.of(manifestBuilder.build());
     }
+
+    public static Map<String, String> updatePartitionMetadataWithFileNamesAndSizes(PartitionUpdate partitionUpdate, Map<String, String> metadata)
+    {
+        ImmutableMap.Builder<String, String> partitionMetadata = ImmutableMap.builder();
+        List<FileWriteInfo> fileWriteInfos = new ArrayList<>(partitionUpdate.getFileWriteInfos());
+
+        // Sort the file infos based on fileName
+        fileWriteInfos.sort(Comparator.comparing(info -> Integer.valueOf(info.getWriteFileName())));
+
+        // Join the file names into a consolidated string
+        String fileNames = joinFileNames(fileWriteInfos.stream().map(FileWriteInfo::getWriteFileName).collect(toImmutableList()));
+
+        // Join the file sizes
+        String fileSizes = Joiner.on(COMMA).join(fileWriteInfos.stream()
+                .map(FileWriteInfo::getFileSize)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(String::valueOf)
+                .collect(toImmutableList()));
+
+        partitionMetadata.put(FILE_NAMES, fileNames);
+        partitionMetadata.put(FILE_SIZES, fileSizes);
+        partitionMetadata.putAll(metadata);
```

Comment:
Leaning towards having a version (e.g., "V1", "V2", ...) for the file names/sizes format. This can be helpful in the future to use different serde. Once we have that, we can put `joinFileNames` and Joiner on sizes into for example a V1 class for serde processing. We may introduce V2 sometime in the future.